supernode/daemon/mgr/gc/gc_task.go

```diff
@@ -0,0 +1,73 @@
+package gc
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+func (gcm *Manager) gcTasks(ctx context.Context) {
+	var removedTaskCount int
+
+	// get all taskIDs and the corresponding accessTime
+	taskAccessMap, err := gcm.taskMgr.GetAccessTime(ctx)
+	if err != nil {
+		logrus.Errorf("failed to get task accessTime map for GC: %v", err)
+		return
+	}
+
+	// range all tasks and determine whether they are expired
+	taskIDs := taskAccessMap.ListKeyAsStringSlice()
+	totalTaskNums := len(taskIDs)
+	for _, taskID := range taskIDs {
+		atime, err := taskAccessMap.GetAsTime(taskID)
+		if err != nil {
+			logrus.Errorf("failed to get access time taskID(%s): %v", taskID, err)
+			continue
+		}
+		if time.Since(atime) < gcm.taskExpireTime {
+			continue
+		}
+
+		if !gcm.gcTask(ctx, taskID) {
+			continue
+		}
+		if err := taskAccessMap.Remove(taskID); err != nil {
+			logrus.Warnf("failed to remove taskID(%s) from access map: %v", taskID, err)
+		}
+		removedTaskCount++
+	}
+
+	logrus.Infof("success to full gc task count(%d), remainder count(%d)", removedTaskCount, totalTaskNums-removedTaskCount)
+}
+
+func (gcm *Manager) gcTask(ctx context.Context, taskID string) bool {
+	// get CIDs according to the taskID
+	cids, err := gcm.dfgetTaskMgr.GetCIDsByTaskID(ctx, taskID)
+	if err != nil {
+		return false
+	}
+
+	return gcm.gcProgessCID(ctx, cids) &&
+		gcm.gcDfgetTaskWithTaskID(ctx, taskID, cids) &&
+		gcm.gcProgessTaskID(ctx, taskID) &&
+		gcm.gcTaskInfo(ctx, taskID)
+}
+
+func (gcm *Manager) gcProgessTaskID(ctx context.Context, taskID string) bool {
+	if err := gcm.progressMgr.DeleteTaskID(ctx, taskID); err != nil {
+		logrus.Errorf("failed to gc progress info taskID(%s): %v", taskID, err)
+		return false
```

Comment:
I found that we return a boolean and record a log message. We all know that a boolean type almost carries no information. So I guess you use a record of log error message to throw put the real information. However, we will find this function does two parts of things: first, return a boolean with little information; second, record a messsage in logfile(because of boolean, then the log record is a must-do). It makes this function do more things which could be simplified.
How about the following way to be pure, and make the caller to do the decision?
```
func (gcm *Manager) gcProgessTaskID(ctx context.Context, taskID string) error {
    return err := gcm.progressMgr.DeleteTaskID(ctx, taskID)
}

```

Reply:
OK, I will try to do that.